api/filters/replacement/replacement_test.go

```diff
@@ -2011,6 +2011,104 @@ spec:
         name: nginx-tagged
       - image: postgres:1.8.0
         name: postgresdb
+`,
+		},
+		"string source -> integer target": {
+			input: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config
+data:
+  PORT: "8080"
+---
+apiVersion: apps/v1
+kind: Pod
+metadata:
+  name: pod
+spec:
+  containers:
+  - image: busybox
+    name: myapp-container
+    ports:
+    - port: 80
+`,
+			replacements: `replacements:
+- source:
+    kind: ConfigMap
+    name: config
+    fieldPath: data.PORT
+  targets:
+  - select:
+      kind: Pod
+    fieldPaths:
+    - spec.containers.0.ports.0.port
```

Comment:
Question, what happens in the scenario that this path does not yet exist and we have to create it? e.g. 

```
replacements:
- source:
    kind: ConfigMap
    name: config
    fieldPath: data.PORT
  targets:
  - select:
      kind: Pod
    fieldPaths:
    - spec.containers.0.ports.0.port
    options:
      create: true
```

Would you mind adding a test for this case? No need to fix the behavior in this PR if it does something undesirable - but it would be good to just have the test as a record of the behavior.

Reply:
Good idea!

Current behavior would be to create it with the source type (since Kustomize doesn't know what the target type is if it didn't already parse it)

Will add!

Reply:
Added.  Though it did expose another problem in the replacement code: Apparently the "create" option doesn't deal well with appending to arrays or creating intermediate maps or arrays.

I'll log separate issues about those.

Reply:
Turns out intermediate maps are fine. Intermediate arrays are complicated.  Following up on that in #4507